core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/stopstartus/StopStartUpscaleActions.java

```diff
@@ -137,17 +160,16 @@ protected void doExecute(StopStartUpscaleContext context, StopStartUpscaleStartI
                 List<InstanceMetadataView> allInstanceMetadata = instanceMetaDataService.getAllAvailableInstanceMetadataViewsByStackId(stack.getId());
                 List<InstanceMetadataView> startedInstancesMetaData = cloudInstanceIdToInstanceMetaDataConverter.getNotDeletedAndNotZombieInstances(
                         allInstanceMetadata, context.getHostGroupName(), cloudInstanceIdsStarted);
-                clusterUpscaleFlowService.instancesStarted(stack.getId(), startedInstancesMetaData);
 
-                handleInstanceUnsuccessfulStart(context, cloudVmInstanceStatusList);
-
-                // This list is currently empty. It could be populated later in another flow-step by querying CM to get service health.
-                // Meant to be a mechanism which detects cloud instances which are RUNNING, but not being utilized (likely due to previous failures)
                 List<CloudInstance> instancesWithServicesNotRunning = payload.getStartInstanceRequest().getStartedInstancesWithServicesNotRunning();
                 List<InstanceMetadataView> metaDataWithServicesNotRunning = cloudInstanceIdToInstanceMetaDataConverter.getNotDeletedAndNotZombieInstances(
                         allInstanceMetadata,
                         context.getHostGroupName(),
-                        instancesWithServicesNotRunning.stream().map(i -> i.getInstanceId()).collect(Collectors.toUnmodifiableSet()));
+                        instancesWithServicesNotRunning.stream().map(CloudInstance::getInstanceId).collect(Collectors.toUnmodifiableSet()));
+
+                clusterUpscaleFlowService.instancesStarted(stack.getId(), startedInstancesMetaData, metaDataWithServicesNotRunning);
+
+                handleInstanceUnsuccessfulStart(context, cloudVmInstanceStatusList);
 
                 LOGGER.info("StartedInstancesCount={}, StartedInstancesMetadataCount={}," +
```

Comment:
Nit: Can this be logged a little earlier? Don't want random issues like null hostnames causing a failure, and then not knowing what we were looking up. i.e. log as early as possible.

Reply:
Updated the order of logging